lib/transfer_jobs/sidekiq_mover.rb

```diff
@@ -105,10 +104,18 @@ def acquire_locks_in_dest(jobs)
       def release_locks_in_source(jobs)
         jobs.each do |job, _|
           locker = Locker.new(job.item, source.redis)
-          locker.lock.unlock(:server)
+          with_sidekiq_redis(source.redis) { locker.lock.unlock(:server) }
         end
       end
 
+      def with_sidekiq_redis(redis)
+        old_redis = ::Sidekiq.redis_pool
+        ::Sidekiq.redis = redis.client.options
```

Comment:
Can you walk me through this? Is this equivalent to our `redis.redis.redis`?

Reply:
The purpose of this is to turn a _redis_ into a redis _pool_. The reason I need to do this dance at all is that sidekiq-unique-jobs ignores the `redis_pool` argument passed[1] and always uses the default sidekiq one. In our case we want to control whether we're looking at the source or destination redis. 

Though looking back at this it appears we'll create a redis connection for every lock! Definitely not desirable. Ideally we should create a single pool for the destination and source redises and always use those.

[1] https://github.com/mhenrixon/sidekiq-unique-jobs/blob/v5.0.10/lib/sidekiq_unique_jobs/scripts/acquire_lock.rb#L15